iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java

```diff
@@ -152,31 +147,136 @@ public boolean hasProperty(String key) {
     }
 
     @Override
-    public String getProperty(String key) {
-        Object val = properties.get(key);
-        return val != null ? val.toString() : null;
+    public Set<String> getPropertyNames() {
+        return properties.keySet();
     }
 
     @Override
-    public String getProperty(String key, String defaultValue) {
-        Object val = properties.getOrDefault(key, defaultValue);
-        return val != null ? val.toString() : null;
+    public PropertyType getPropertyType(String key) {
+        if (properties.containsKey(key)) {
+            return properties.get(key).propertyType();
+        }
+        return null;
```

Comment:
Not sure if it should return null or throw an exception (what do you think @annetill ?) In any case, the behavior should be described in the javadoc of the `Identifiable` API